### ISSUE TITLE
Add evolutionary tracks as a 2nd option for generating the stellar grid

### ISSUE THREAD

**Issue by karllark** [BEAST-Fitting/beast#18]
The stellar grid is currently based on isochrones.  This the grid has uniform spacing in age, but non-uniform in mass.  Isochrones are best for dealing with ensembles of stars, but not necessarily for individual stars.  The issue for the BEAST is that the age spacing must be fine enough to sample the evolution of high mass stars and this severely over samples the evolution of low mass stars.  

One solution is to change from using isochrones to evolutionary tracks.  Stellar interior models create the evolutionary tracks which are then interpolated to isochrones.  The tracks are given for a set of stars with a range of initial masses.  Thus the stellar grid created from them will have a uniform spacing in mass, and a non-uniform spacing in age.  This should create a significantly smaller grid with the same effective coverage of stellar parameters.  Smaller grids mean faster run times *or* the ability to create grids with finer sampling of stellar parameters.

Basing the stellar grid on evolutionary tracks is definitely possible as this was what the original version of the BEAST did (in IDL).

Leo Girardi stated in an email to me that he has code for doing efficient/good interpolation on the stellar evolutionary tracks.  Hopefully, this is in python.

**Comment by karllark** [comment]
Working on other tasks, I have found that adding in the evolutionary tracks should be fairly straightforward.  Need to provide a new function that gets the tracks, interpolates them to the desired uniform mass spacing, and provides the needed 'pts' (e.g., logg, logTeff, etc.) for use in generating the grid.
The only additional change is to handle the switch to non-uniform spacing in age instead of mass.  This mainly affects the computation of the stellar priors and (possibly) the 1d pPDFs.

**Comment by SteveGoldmanSTSCI** [comment]
I created a new fits file which includes interpolated Padova tracks from Phil Rosenfield (http://philrosenfield.github.io/padova_tracks/). I have been able to import this file in model_grid.py where the star model library is imported on line 66. This was just to test to see if the beast would run with the new file, full well knowing the end results would be incorrect. More work to come.

**Comment by s-goldman** [comment]
I have a table (fits) of all the Padova data including the eeps (evolutionary points), the index, and the metallicities. Now working on creating an N-dimensional mesh grid to interpolate over. What we currently have is something like this:

![karls_plot1](https://user-images.githubusercontent.com/32876747/39383757-7317ea82-4a38-11e8-8a11-1302c88c765c.png)

but we need it evenly sampled.

**Comment by karllark** [comment]
Why does the mass change on a single track?  Are you plotting the initial or current mass?  The interpolation should be done in initial mass.

**Comment by future-steve** [comment]
Yeah the models only have initial masses, I think this was just all the models for a given metallicity. Not sure about the squiggle to the right of everything though.

**Comment by s-goldman** [comment]
New plan, I am downloading the full Parsec evolutionary tracks and interpolating them. In the end I should have an finely sampled HR diagram of initial mass and age.

**Comment by s-goldman** [comment]
new new plan, working on creating grid from the full Padova models where each evolutionary track only has 20 points (in intervals of 5%). Doing interpolation over indices to the 5% intervals and then interpolation of each column of the tracks.

**Comment by karllark** [comment]
The 20 points we used in the example today was only that, an example.  The number of points per evolutionary track will vary depending on how after the luminosity, etc. changes.  Right?

**Comment by karllark** [comment]
Potential way to interface to evolutionary tracks.  Looks to only support MIST tracks fro now. 
 https://github.com/timothydmorton/isochrones
